docs/ocis/adr/0008-policy-enforcement.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "8. Extensions Policies"
+date: 2021-06-30T14:00:00+01:00
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/ocis/adr
+geekdocFilePath: 0008-policy-enforcement
+---
+
+* Status: proposed
+* Deciders: @butonic, @micbar, @dragotin, @hodyroff, @pmaier1, @fschade
+* Date: 2021-06-30
+
+## Context and Problem Statement
+
+There should be a way to impose certain limitations in areas of the code that require licensing. This document researches an approach to achieve it.
+
+## Decision Drivers
+
+- as a team, we want to have the licensing code concentrated in a central module
+- we don't want to stop/start the extension whenever a policy is updated (hot reload)
+
+## Considered Options
+
+1. Build the evaluation engine in-house.
+2. Use third party libraries such as Open Policy Agent (a CNCF aproved project with a Go library)
+
+## Decision Outcome
+
+Chosen option: option 2; Use third party libraries such as Open Policy Agent (a CNCF aproved project with a Go library)
+
+### Positive Consequences
+
+- OPA is production battle tested.
+- built around performance - policies evaluations are no longer than 1ms per request.
+- middleware friendly: we use gRPC clients all over our ecosystem; wrappers (or middlewares) is a viable way to solve this problem instead of a dedicated service or its own package.
+- community support.
+- kubernetes friendly.
```

Comment:
maybe worth to mention that also envoy, kong, terraform, traefik, php, node... many more are supported